### PR TITLE
UIF-442: Need new permission(s) to view Finance settings in UI

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UIF-442](https://issues.folio.org/browse/UIF-442) - add permissions and disable action buttons
## Approach

https://github.com/folio-org/ui-finance/assets/116072773/0670bc30-08a0-4eef-b3dd-2c135476d782

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by alisher-epam** [comment]
The video source shows how it works. However, the example isn't complete due to don't have access to update permissions. It needs to be updated from thendatabase to check it locally.

File: package.json [hunk 1]
```diff
@@ -60,6 +60,31 @@
         "displayName": "UI: Finance module is enabled",
         "visible": false
       },
+      {
+        "permissionName": "ui-finance.settings.view",
+        "displayName": "Settings (Finance): View settings",
```
Comment (Terala-Priyanka) [inline package.json:65]: Please add translations to the new permission name.

File: package.json [hunk 2]
```diff
@@ -60,6 +60,31 @@
...
+          "finance.expense-classes.collection.get",
+          "finance.fiscal-years.collection.get",
+          "finance.fund-codes-expense-classes.collection.get",
+          "finance.fund-types.collection.get",
+          "users.collection.get"
+        ]
+      },
+      {
+        "permissionName": "ui-finance.settings.exportFundAndExpenseClassCodes",
+        "displayName": "Settings (Finance): Export fund and expense class codes",
```
Comment (Terala-Priyanka) [inline package.json:78]: Please add translations to the new permission name.

**Comment by usavkov-epam** [comment]
You also have to support i18n for new perms

File: package.json [hunk 3]
```diff
@@ -119,6 +144,7 @@
         "visible": true,
         "subPermissions": [
           "settings.finance.enabled",
+          "ui-finance.settings.view",
```
Comment (Terala-Priyanka) [inline package.json:147]: I think you can remove the below permissions in the list of sub-permissions when you add "ui-finance.settings.view" here, as they will be inherited.

"settings.finance.enabled",
          "finance.expense-classes.collection.get",
          "finance.fiscal-years.collection.get",
          "finance.fund-codes-expense-classes.collection.get",
          "finance.fund-types.collection.get",
          "users.collection.get"

File: src/settings/FundTypeSettings/FundTypeSettings.js [hunk 4]
```diff
@@ -40,6 +42,11 @@ const FundTypeSettings = ({
       hiddenFields={hiddenFields}
       nameKey="name"
       id="fundTypes"
+      canCreate={hasEditPerms}
+      actionSuppressor={{
+        edit: () => !hasEditPerms,
+        delete: () => !hasEditPerms,
```
Comment (Terala-Priyanka) [inline src/settings/FundTypeSettings/FundTypeSettings.js:48]: you should frame the const hasEditPerms based on the specific PUT permission.
make another declaration for hasDeletePerms based on the specific DELET permission.
hook them accordingly to the actionSuppressor

File: src/settings/ExpenseClassSettings/ExpenseClassSettings.js [hunk 5]
```diff
@@ -50,6 +51,11 @@ const ExpenseClassSettings = ({ stripes }) => {
       visibleFields={visibleFields}
       rowFilter={<ExpenseClassHelper />}
       validate={validate}
+      canCreate={hasEditPerms}
+      actionSuppressor={{
+        edit: () => !hasEditPerms,
+        delete: () => !hasEditPerms,
```
Comment (Terala-Priyanka) [inline src/settings/ExpenseClassSettings/ExpenseClassSettings.js:57]: you should frame the const hasEditPerms based on the specific PUT permission.
make another declaration for hasDeletePerms based on the specific DELET permission.
hook them accordingly to the actionSuppressor

**Review by Terala-Priyanka — CHANGES_REQUESTED** [review]

**Review by Terala-Priyanka — APPROVED** [review]

**Comment by alisher-epam** [comment]
[This PR](https://github.com/folio-org/ui-finance/pull/754) will resolve failing Github-actions https://github.com/folio-org/ui-finance/pull/754

**Comment by alisher-epam** [comment]
@usavkov-epam please take a look at this PR when you have a chance please.

File: src/settings/ExpenseClassSettings/ExpenseClassSettings.test.js [hunk 6]
```diff
@@ -17,12 +18,17 @@ jest.mock('@folio/stripes-smart-components/lib/ControlledVocab', () => {
   return MockComponent;
 });
 
-const renderExpenseClassSettings = () => render(<ExpenseClassSettings stripes={{}} />);
+const stripesMock = { hasPerm: jest.fn(() => true) };
```
Comment (mariia-aloshyna) [inline src/settings/ExpenseClassSettings/ExpenseClassSettings.test.js:21]: if you are not going to test `hasPerm` method then it should be just func - `() => true`, jest mock is not needed here

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**